packages/hooks/src/hooks.ts

```diff
@@ -68,3 +68,23 @@ export function useElementNode<ElementType = HTMLElement>() {
 
   return [element, elementRefCallback] as const;
 }
+
+export function useDocumentEventListener(
```

Comment:
Could we add TSDoc annotation to these since they're soon to be externally consumable? Would also be nice for our own development 😬

Also, we should probably have tests for these hooks.

Reply:
May want to use this: https://github.com/mpeyper/react-hooks-testing-library

Reply:
Added TSDoc annotation! Hopping over to menu but will work on adding tests shortly :)